src/texttransformation.js

```diff
@@ -0,0 +1,309 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module typing/texttransformation
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import TextWatcher from '@ckeditor/ckeditor5-utils/src/textwatcher';
+
+// All named transformations.
+const TRANSFORMATIONS = {
+	// Common symbols:
+	copyright: { from: '(c)', to: '©' },
+	registered_trademark: { from: '(r)', to: '®' },
```

Comment:
Why underscored? Unless there's some reason, as all token names they should follow the camelCase.